src/legacy/state/orders/hooks.ts

```diff
@@ -408,3 +410,63 @@ export const useSetIsOrderRefundedBatch = (): SetIsOrderRefundedBatchCallback =>
   const dispatch = useDispatch<AppDispatch>()
   return useCallback((params: SetIsOrderRefundedBatch) => dispatch(setIsOrderRefundedBatch(params)), [dispatch])
 }
+
+/**
+ * Hook that "cleans" orders local storage
+ * Related issue https://github.com/cowprotocol/cowswap/issues/2690
+ *
+ * It runs when the parent component is unmounted
+ */
+export const useCleanOrdersStorageOnUnmount = () => {
+  useEffect(() => {
+    return () => {
+      // Get orders state from local storage
+      const ordersCache = localStorage.getItem(ORDER_STORAGE_KEY)
+
+      if (!ordersCache) {
+        return
+      }
+
+      const parsedState = JSON.parse(ordersCache)
```

Comment:
Is it a good idea to mutate the storage managed by redux outside of it?
Couldn't this cause race conditions with redux updating the data?

Reply:
My idea was that it will be good enough if this happens successfully at least some times when the user is closing the app and then the cache will be cleaned and it won't grow too much. What do you think?

Reply:
I like your approach and how you are selecting the orders to keep.
Just worried of potential race conditions with redux.

Would this be possible to be run as a redux action?

For example, you can use the exact same logic you have right now, triggering/dispatching on unmount, but instead of hacking your way through the local storage key, you add one new redux action that runs this logic.

Reply:
I've changed the implementation to use redux